src/BatteryStateOfChargeService.cpp

```diff
@@ -16,21 +23,108 @@ BatteryStateOfChargeService::~BatteryStateOfChargeService()
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+    return BATTERY_AMP_HOUR_CAPACITY-ampHours ;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
+    if(newCurrent<0)
+    {
     return false;
+    }
+    else
+    {
+       return true;
+    }
 }
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 {
-    return QTime::currentTime();
+    return t;
```

Comment:
The time left when charged or depleted only needs to known once this function is called, so it would also make sense to calculate this value in this function. You're currently doing it `addData()`, that function is very large and moving some of the functionality to this function would also increase readability. Finding this value in `addData()` also created the cryptic `Prime` word in your variables. 

By moving the functionality into this function, you would be solving more than one issue :)